enterprise/cli/server_dbcrypt.go

```diff
@@ -0,0 +1,331 @@
+//go:build !slim
+
+package cli
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+	"strings"
+
+	"cdr.dev/slog"
+	"cdr.dev/slog/sloggers/sloghuman"
+	"github.com/coder/coder/v2/cli"
+	"github.com/coder/coder/v2/cli/clibase"
+	"github.com/coder/coder/v2/cli/cliui"
+	"github.com/coder/coder/v2/enterprise/dbcrypt"
+
+	"golang.org/x/xerrors"
+)
+
+func (r *RootCmd) dbcryptCmd() *clibase.Cmd {
+	dbcryptCmd := &clibase.Cmd{
+		Use:   "dbcrypt",
+		Short: "Manage database encryption.",
+		Handler: func(inv *clibase.Invocation) error {
+			return inv.Command.HelpHandler(inv)
+		},
+	}
+	dbcryptCmd.AddSubcommands(
+		r.dbcryptDecryptCmd(),
+		r.dbcryptDeleteCmd(),
+		r.dbcryptRotateCmd(),
+	)
+	return dbcryptCmd
+}
+
+func (*RootCmd) dbcryptRotateCmd() *clibase.Cmd {
+	var flags rotateFlags
+	cmd := &clibase.Cmd{
+		Use:   "rotate",
+		Short: "Rotate database encryption keys.",
+		Handler: func(inv *clibase.Invocation) error {
+			ctx, cancel := context.WithCancel(inv.Context())
+			defer cancel()
+			logger := slog.Make(sloghuman.Sink(inv.Stdout))
+			if ok, _ := inv.ParsedFlags().GetBool("verbose"); ok {
+				logger = logger.Leveled(slog.LevelDebug)
+			}
+
+			if err := flags.valid(); err != nil {
+				return err
+			}
+
+			ks := [][]byte{}
+			dk, err := base64.StdEncoding.DecodeString(flags.New)
+			if err != nil {
+				return xerrors.Errorf("decode new key: %w", err)
+			}
+			ks = append(ks, dk)
+
+			for _, k := range flags.Old {
+				dk, err := base64.StdEncoding.DecodeString(k)
+				if err != nil {
+					return xerrors.Errorf("decode old key: %w", err)
+				}
+				ks = append(ks, dk)
+			}
+
+			ciphers, err := dbcrypt.NewCiphers(ks...)
+			if err != nil {
+				return xerrors.Errorf("create ciphers: %w", err)
+			}
+
+			newDigest := ciphers[0].HexDigest()
+			oldDigests := make([]string, 0, len(ciphers)-1)
+			for _, c := range ciphers[1:] {
+				oldDigests = append(oldDigests, c.HexDigest())
+			}
+			if len(oldDigests) == 0 {
+				oldDigests = append(oldDigests, "none")
+			}
+			msg := fmt.Sprintf(`Rotate external token encryptions keys?\n- New key: %s\n- Old keys: %s`,
```

Comment:
You should probably print the keys here unless a "no-print-key" flag was supplied or something. The digests are not very helpful and can't be compared to the actual keys the user has very easily

Reply:
I can print the base64-encoded keys passed to the CLI, sure.
The `--yes` option probably counts as a `no-print-key` flag implicitly.